File: LeetCode/LeetCode Practice Problems/657- Robot Return to Origin.cpp
class Solution {
public:
    bool judgeCircle(string moves) {
        int upDown = 0 , leftRight = 0;
        for(char c : moves)
        {
            if(c == 'U')
                upDown++;
            else if(c == 'D')
                upDown--;
            else if(c == 'L')
                leftRight++;
            else
                leftRight--;
        }
        
        return (upDown == 0)&&(leftRight == 0);
    }
};

File: LeetCode/LeetCode Practice Problems/657-Robot Return to Origin.java
class Solution {
    public boolean judgeCircle(String moves) {
        int c1 = 0 , c2 = 0;
        
        for(char c: moves.toCharArray()) {
            if(c == 'U') c1++;
            else if(c == 'D') c1--;
            else if(c == 'L') c2++;
            else c2--;
        }
        
        return c1 == 0 && c2 == 0;
    }
}
